zerver/lib/scheduled_messages.py

```diff
@@ -19,14 +22,20 @@ def further_validated_scheduled_message_dict(
     for which can be used to directly create a Draft object."""
 
     content = normalize_body(draft_dict["content"])
+    
+    print('content: ' + content)
 
-    timestamp = draft_dict.get("timestamp", time.time())
-    timestamp = round(timestamp, 6)
-    if timestamp < 0:
-        # While it's not exactly an invalid timestamp, it's not something
-        # we want to allow either.
-        raise JsonableError(_("Timestamp must not be negative."))
-    last_edit_time = timestamp_to_datetime(timestamp)
+    timestamp = draft_dict.get("deliver_at")
+    if re.match(r".* [1-9] .*", timestamp):
+      print('TEST')
+    # timestamp = round(timestamp, 6)
+    # if timestamp < 0:
+    #     # While it's not exactly an invalid timestamp, it's not something
+    #     # we want to allow either.
+    #     raise JsonableError(_("Timestamp must not be negative."))
+    deliver_at = datetime.strptime(timestamp, '%b %d %Y %I:%M%p')
```

Comment:
Because `last_edit_time` was a datetime and not a string, I'm guessing it would be better to store this as a datetime as well. Maybe the reason for this is to make it easier to query/sort by time :)